### DISCUSSION TITLE
Bulk conversion of SMILES codes to JME format

### DISCUSSION THREAD

**Discussion by Conceptsinchem** [jsme-editor/jsme-editor.github.io#8] [General]
Hi,

Just want to say thank you for the software, the JSME editor has been a very useful tool for our chemistry courses.

We have databases of SMILES codes as answers to chemistry drawing problems, however we'd like to be able to display the correct answers as the structure image (we use LON-CAPA). I need to convert the SMILES codes to the JME output format for the molecule display. 

I've done this manually for some smaller databases, but was wondering, is there was a way to run a script to do this automatically in bulk through the JSME? 

Cheers
Mark

**Comment by jsme-editor** [comment]
Hi,

we don't have a script that can convert SMILES codes to the JME output format.  But why the JME output format if I may ask? MOL should work as well and there are many tools that provide conversion from SMILES to MOL.

FYI, I think that the JME file format is implemented in the CACTVS toolkit (https://www.xemistry.com/).


Bruno

**Reply by Conceptsinchem** [reply]
Thanks for the reply. We're using LON-CAPA to deliver content to our students and JME format is required to display the molecule as an image using the <organicstructure> tag, e.g;

organicstructure molecule="4 3 C 6.33 -6.22 C 7.73 -6.22 O 8.43 -7.43 O 8.43 -5.00 1 2 1 2 3 1 2 4 2" 

As well as in the optional jmeanswer parameter to display the answer to the students as an image. We have a large database of questions which include SMILES codes and would like to convert them to JME in an automated fashion.

Cheers,
Mark
